icu4c/source/common/unicode/umachine.h

```diff
@@ -52,6 +52,16 @@
 #include <stdbool.h>
 #include <stddef.h>
 
+#if U_CPLUSPLUS_VERSION == 0
+#if !U_PLATFORM_IS_DARWIN_BASED
+#include <uchar.h>
+#else
+// The <uchar.h> file does not exist in Apple SDKs (as of 3/15/24),
+// so we have to define char16_t ourselves
+typedef uint16_t char16_t;
+#endif // !U_PLATFORM_IS_DARWIN_BASED
+#endif // U_CPLUSPLUS_VERSION == 0
+
```

Comment:
optional: It feels like this might fit better in ptypes.h, sometime after it includes platform.h.
Looking at the various header files,
- platform.h tells us about the platform
- ptypes.h levels out platform differences -- recently most of it was removed since we rely on C++17 and C11 (ICU-22612)
- umachine.h defines ICU-specific types on top of the previous headers

But we can move this snippet later if you like.

Reply:
Nah, I went ahead and moved it.  I think you're right.